test/metabase/test/util/log.clj

```diff
@@ -2,12 +2,12 @@
   "Utils for controlling the logging that goes on when running tests."
   (:import java.io.PrintStream
            [org.apache.commons.io.output NullOutputStream NullWriter]
-           [org.apache.log4j Level Logger LogManager]))
+           [org.apache.logging.log4j Level LogManager]
+           [org.apache.logging.log4j.core Logger LoggerContext]))
 
 (def ^:private logger->original-level
-  (let [loggers (cons
-                 (Logger/getRootLogger)
-                 (enumeration-seq (LogManager/getCurrentLoggers)))]
+  (let [loggers (.. ^LoggerContext (LogManager/getContext false)
+                    (getLoggers))]
```

Comment:
Don't worry about changing this but:

1. `..` works like `->` or `->>`, so you don't need to wrap `(getLoggers)` in parens 
2. I think using `..` with only 2 args is less readable than just doing `(.getLoggers (LogManager/getContext false))`
3.  I'm 95% sure you don't need that type hint there

Reply:
The type hint is necessary because there are two `LoggerContext`s in log4j2, one is an interface from the base API, and the other is the concrete implementation, which we cast / hint to so that we can call methods that are part of the impl.

It's a weird thing about how they made the API. See https://logging.apache.org/log4j/2.x/manual/customconfig.html#AddingToCurrent